docs/developers/napari_architecture.md

```diff
@@ -0,0 +1,164 @@
+# Napari architecture
```

Comment:
I really like this opening section. It sets up the rest of the document well. Maybe a small/simple diagram that depicts some of the organization described below would be useful?